### PR TITLE
[FFI] Remove all code & test suites tagged with J9VM_OPT_PANAMA

### PR DESCRIPTION
**Pull request by ChengJin01**
The changes remove the stale code & tests tagged with `J9VM_OPT_PANAMA`
in JDK16 given all FFI related functionalities were fully implemented
in which case the code & test surrounded by or related to 
`J9VM_OPT_PANAMA` are no longer required for any reference.

Signed-off-by: ChengJin01 <jincheng@ca.ibm.com>

Fixes: #10377.

### CODE REVIEW

**Comment by ChengJin01** [comment]
The PR has been verified in personal builds (internal & open) on all supported platforms.

Reviewer: @keithc-ca 
FYI: @tajila, @pshipton

**Comment by keithc-ca** [comment]
Jenkins test sanity amac jdk11,jdk23

File: runtime/vm/MHInterpreter.inc [hunk 1]
```diff
@@ -59,11 +59,7 @@ static const char * const names[] = {
 	"J9_METHOD_HANDLE_KIND_FILTER_ARGUMENTS",
 	"J9_METHOD_HANDLE_KIND_VARHANDLE_INVOKE_EXACT",
 	"J9_METHOD_HANDLE_KIND_VARHANDLE_INVOKE_GENERIC",
-#if defined(J9VM_OPT_PANAMA)
-	"J9_METHOD_HANDLE_KIND_NATIVE",
-#else /* defined(J9VM_OPT_PANAMA) */
 	"",
```
Comment (keithc-ca) [inline runtime/vm/MHInterpreter.inc:62]: Please remove this (blank) entry corresponding to `J9_METHOD_HANDLE_KIND_NATIVE` which has been removed.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/vm/MHInterpreter.inc (hunk 1 above)
Comment (ChengJin01) [reply runtime/vm/MHInterpreter.inc]: Fixed.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**